python/lsst/analysis/drp/dataSelectors.py

```diff
@@ -0,0 +1,136 @@
+__all__ = ("HighSnSelector", "LowSnSelector", "MainFlagSelector", "PsfFlagSelector", "StarSelector",
+           "GalaxySelector")
+
+from lsst.pex.config import ListField, Field
+from lsst.pipe.tasks.dataFrameActions import DataFrameAction
+import numpy as np
+
+
+class FlagSelector(DataFrameAction):
+    """The main set of flags to use to select valid sources for QA"""
+    filterColumns = ListField(doc="Names of the flag columns to filter on", dtype=str, optional=False,
+                              default=["base_PixelFlags_flag"])
+
+    @property
+    def columns(self):
+        yield from self.filterColumns
+
+    def __call__(self, df, **kwargs):
+        """Select on the given flags"""
+        result = None
+        for flag in self.filterColumns:
+            selected = (df[flag] == 0)
+            if result is None:
+                result = selected
+            else:
+                result &= selected
+        return result
+
+
+class MainFlagSelector(FlagSelector):
+    """The main set of flags to use to select valid sources for QA"""
+
+    bands = ListField(doc="The bands to apply the flags in",
+                      dtype=str,
+                      default=["g", "r", "i", "z", "y"])
+
+    def setDefaults(self):
+        super().setDefaults()
+        flagCols = ["PixelFlags", "Blendedness_flag"]
+        filterColumns = [band + flag for flag in flagCols for band in self.bands]
+        self.filterColumns = filterColumns
+
+
+class PsfFlagSelector(FlagSelector):
```

Comment:
The selectors included in this ticket are primarily included as examples and I expect them all to be updated/replaced on future tickets once we start making actual QA plots rather than just example plots.